applications/GeoMechanicsApplication/custom_workflows/dgeoflow.h

```diff
@@ -46,56 +46,109 @@
+                    minCriticalHead(minCriticalHead), maxCriticalHead(maxCriticalHead), stepCriticalHead(stepCriticalHead)
+            {}
+        };
+
+        struct CallBackFunctions
+        {
+            std::function<void(const char*)> rLogCallback;
+            std::function<void(double)> rReportProgress;
+            std::function<void(const char*)> rReportTextualProgress;
+            std::function<bool()> rShouldCancel;
+
+            CallBackFunctions(std::function<void(const char*)> rLogCallback,
+                              std::function<void(double)> rReportProgress,
+                              std::function<void(const char*)> rReportTextualProgress,
+                              std::function<bool()> rShouldCancel) :
+                    rLogCallback(std::move(rLogCallback)), rReportProgress(std::move(rReportProgress)), rReportTextualProgress(std::move(rReportTextualProgress)),
+                    rShouldCancel(std::move(rShouldCancel))
+            {}
+        };
+
+        int ExecuteFlowAnalysis(std::string_view rWorkingDirectory,
+                                const std::string& rProjectParamsFileName,
+                                const CriticalHeadInfo& criticalHeadInfo,
+                                std::string_view rCriticalHeadBoundaryModelPartName,
+                                const CallBackFunctions& rCallBackFunctions);
+
+        void ExecuteWithoutPiping(ModelPart& model_part,
+                                  const Kratos::Parameters& gid_output_settings) const;
+
+        int ExecuteWithPiping(ModelPart& model_part,
+                              const Kratos::Parameters& gid_output_settings,
+                              const CriticalHeadInfo& criticalHeadInfo,
+                              LoggerOutput::Pointer p_output,
+                              const CallBackFunctions& rCallBackFunctions);
+
+        void AddNodalSolutionStepVariables(ModelPart& model_part) const;
+
+        int FindCriticalHead(ModelPart& model_part,
+                               const Kratos::Parameters& gid_output_settings,
+                               const CriticalHeadInfo& criticalHeadInfo,
+                               LoggerOutput::Pointer p_output,
+                               const shared_ptr<Process>& RiverBoundary,
+                               const GeoMechanicsNewtonRaphsonErosionProcessStrategyType::Pointer& p_solving_strategy,
+                               const CallBackFunctions& rCallBackFunctions);
+
+        void HandleCriticalHeadFound(const CriticalHeadInfo& criticalHeadInfo);
+
+        void HandleCancellationAndReset(const std::function<void(const char*)>& rLogCallback,
+                                    LoggerOutput::Pointer p_output);
+
 
     private:
         // Initial Setup
         Model current_model;
         Kernel kernel;
         KratosGeoMechanicsApplication::Pointer geoApp;
```

Comment:
These members do not yet adhere to the style guide
```suggestion
        Model mCurrentModel;
        Kernel mKernel;
        KratosGeoMechanicsApplication::Pointer mpGeoApp;
```